### PR TITLE
FreeBSD compatability [WIP]

### PR DESCRIPTION
**Pull request by dennisdegreef**
Can I get some feedback on the work i've been doing on freebsd compatability?

Tasks:
- [X] Configurable etc-directory for `/etc` and `/usr/local/etc`
- [X] Configurable root-group, linux has `root`, bsd has `wheel`
- [X] rc-scripts in `/etc/rc.conf.d/`
- [X] Create separate `/usr/local/etc/rc.d/openvpn-${name}` rc-scripts from `openvpn::server`
- [X] Change `/etc/rc.conf.d/openvpn` to `/etc/rc.conf.d/openvpn-${name}`
- [ ] Fix up tests again

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -100,6 +100,13 @@
         }
       }
     }
+    'FreeBSD': {
+      $group = 'nogroup'
```
Comment (dennisdegreef) [inline manifests/params.pp]: Align properly

File: manifests/init.pp [hunk 2]
```diff
@@ -43,6 +43,16 @@
   $autostart_all = true,
 ) {
 
+  $etc_directory = $::operatingsystem ? {
```
Comment (luxflux) [inline manifests/init.pp]: Could you move this directive into `openvpn::params`, wo we have these OS specific thing all at one place.

File: manifests/init.pp [hunk 3]
```diff
@@ -43,6 +43,16 @@
...
+    /(?i:FreeBSD)/ => '/usr/local/etc',
+    default        => '/etc',
+  }
+
+  $root_group = $::operatingsystem ? {
```
Comment (luxflux) [inline manifests/init.pp]: Could you move this directive into `openvpn::params`, wo we have these OS specific thing all at one place.

**Comment by luxflux** [comment]
Hey cool! Thanks for you work! :beers: 

Looks good so far, I added some small comments.

**Comment by dennisdegreef** [comment]
I can't really tell what makes the build fail, could be my lack of 'spec' experience... @luxflux Could you help me out? I may have broken something...

**Comment by luxflux** [comment]
Well, I'm pretty sure that is related to the top scope variable warnings. You should not use `$load_main_service`, better specify the full scope: `$::openvpn::params::load_main_service`. I think then it will work out.

**Comment by luxflux** [comment]
Closed due to inactivity.

**Pull request closed without merging**